op-batcher/batcher/config.go

```diff
@@ -135,18 +136,19 @@ func (c *CLIConfig) Check() error {
 	if !derive.ValidCompressionAlgo(c.CompressionAlgo) {
 		return fmt.Errorf("invalid compression algo %v", c.CompressionAlgo)
 	}
-	if c.BatchType > 1 {
+	if c.BatchType > derive.SpanBatchType {
 		return fmt.Errorf("unknown batch type: %v", c.BatchType)
 	}
 	if c.CheckRecentTxsDepth > 128 {
 		return fmt.Errorf("CheckRecentTxsDepth cannot be set higher than 128: %v", c.CheckRecentTxsDepth)
 	}
-	if c.DataAvailabilityType == flags.BlobsType && c.TargetNumFrames > 6 {
-		return errors.New("too many frames for blob transactions, max 6")
-	}
 	if !flags.ValidDataAvailabilityType(c.DataAvailabilityType) {
 		return fmt.Errorf("unknown data availability type: %q", c.DataAvailabilityType)
 	}
+	// we want to enforce it for both blobs and auto
+	if c.DataAvailabilityType != flags.CalldataType && c.TargetNumFrames > params.MaxBlobGasPerBlock/params.BlobTxBlobGasPerBlob {
+		return fmt.Errorf("too many frames for blob transactions, max %d", params.MaxBlobGasPerBlock/params.BlobTxBlobGasPerBlob)
```

Comment:
The `c.DataAvailabilityType != flags.CalldataType` change is good, but not sure if `params.MaxBlobGasPerBlock/params.BlobTxBlobGasPerBlob` is better than `6`... Maybe we add a `const MaxBlobsPerBlobTx = 6` to the `eth` package a use it here?

Reply:
This is the same style that EL is using: https://github.com/ethereum-optimism/op-geth/blob/5f7ebba8a124ae87225f81a1a9c827f8a534f2b7/core/txpool/validation.go#L162 . And when in future, when the number of supported blobs per block increase, it'll be correct automatically.

Reply:
I see. I still find it clunky to remember every time to use `params.MaxBlobGasPerBlock/params.BlobTxBlobGasPerBlob` instead of `6`. Wdyt about adding
```go
const MaxBlobsPerBlobTx = params.MaxBlobGasPerBlock/params.BlobTxBlobGasPerBlob
```
to package `eth` and then using it here and other places where we use `6`? 

Would also be great if you could scan the batcher package for other occurrences of the magic `6` and then replace it by this new constant. 🙏🏻

Reply:
Done.